js/ui/selection/selection.strategy.js

```diff
@@ -69,8 +69,28 @@ export default Class.inherit({
         return this.selectedItemKeys(keys, preserve, isDeselect, isSelectAll);
     },
 
+    _removeTemplateProp: function(filterItem) {
+        if(isObject(filterItem) && Object.prototype.hasOwnProperty.call(filterItem, 'template')) {
+            delete filterItem.template;
+        }
+    },
+
+    _prepareFilterValue: function(remoteFilter) {
```

Comment:
I think we should name this function more concrete. For example, we can name it `_removeTemplateProperty` and union this 2 functions